site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/SupportForm.js

```diff
@@ -0,0 +1,290 @@
+    withCredentials: true,
+    xsrfCookieName: "csrftoken",
+    xsrfHeaderName: "X-CSRFToken",
+    headers: {
+        "Accept": "application/json",
+        'Content-Type': 'multipart/form-data',
+    },
+};
+
+const http = axios.create(apiConfig);
+
+function formikToFormData(values) {
+    const formData = new FormData();
+    // Append each key/value pair to formData
+    Object.keys(values).forEach((key) => {
+        let currentValue = values[key];
+        if (Array.isArray(currentValue)) {
+            currentValue.forEach((val) => {
+                formData.append(key, val);
+            })
+        } else {
+            formData.append(key, currentValue);
+        }
+    })
+
+    return formData;
+}
+
+
+const FileUploader = ({ dropzoneParams, maxFileSize, name, currentFiles, handleDelete }) => {
+    return (
+        <Dropzone {...dropzoneParams}>
+            {({ getRootProps, getInputProps }) => {
+                const inpProps = {
+                    ...getInputProps(),
+                    // Display the dropzone input, otherwise it's hidden by default.
+                    style: {
+                        'display': 'block'
+                    },
+                    name: name
+                }
+                return (
+                    <div {...getRootProps()}>
+                        <input {...inpProps} />
+                        <label className="helptext mt-0">
+                            {`Optional. Max attachments size:`} {humanReadableBytes(maxFileSize, false)}
+                        </label>
+                        {currentFiles.length > 0
+                            &&
+                            <FileTable
+                                handleDelete={handleDelete}
+                                filesList={currentFiles}
+                            />
+                        }
+                    </div>
+                )
+            }}
+        </Dropzone>
+    );
+};
+
+FileUploader.propTypes = {
+    dropzoneParams: PropTypes.object.isRequired,
+    maxFileSize: PropTypes.number.isRequired,
+    name: PropTypes.string.isRequired,
+    currentFiles: PropTypes.array.isRequired
+};
+
+class SupportForm extends Component {
+
+    constructor(props) {
+        super(props);
+        this.FormSchema = Yup.object({
+            name: Yup.string().required(),
+            email: Yup.string().email().required(),
+            category: Yup.string().required(),
+            subject: Yup.string().required(),
+            description: Yup.string().min(
+                this.props.descriptionMinLength,
+                `Field must be at least ${this.props.descriptionMinLength} characters long.`
+            ).max(
+                this.props.descriptionMaxLength,
+                `Field must be at most ${this.props.descriptionMaxLength} characters long.`)
+                .required(),
+            sysInfo: Yup.boolean()
+        });
+
+        this.state = {
+            files: []
```

Comment:
Files are only validated by `react-dropzone` and no error/validation message is shown. Needs to be implemented.